pkg/storage/pebble.go

```diff
@@ -1187,6 +1198,21 @@ func NewPebble(ctx context.Context, cfg PebbleConfig) (p *Pebble, err error) {
 		opts.FS = encryptionEnv.FS
 	}
 
+	blockLoadLimiter := quotapool.NewIntPool("pebble.read-block-limiter", uint64(BlockLoadConcurrencyLimit.Get(&cfg.Settings.SV)))
+	BlockLoadConcurrencyLimit.SetOnChange(&cfg.Settings.SV, func(ctx context.Context) {
+		blockLoadLimiter.UpdateCapacity(uint64(BlockLoadConcurrencyLimit.Get(&cfg.Settings.SV)))
+	})
+	opts.BeforeReadBlock = func(ctx context.Context) (done func(), _ error) {
```

Comment:
The only thing that stands out here is there is no way to know the number of queued operations are in queue for this semaphore.

Reply:
Isn't that exactly what the metric shows?